### PR TITLE
Include organizations and teams in user profile

### LINKED ISSUE #2903: Add information about org admin, super admin and team memberships to user profiles

### PR DESCRIPTION
**Pull request by JorgeMartinezG**
Closes #2903

### CODE REVIEW

File: backend/api/organisations/resources.py [hunk 1]
```diff
@@ -315,18 +315,6 @@ def get(self):
         except Exception:
             manager_user_id = None
 
-        if manager_user_id is not None:
```
Comment (willemarcel) [inline backend/api/organisations/resources.py:318]: @ramyaragupathy I removed that restriction in order to be able to query the organizations of another user. Is that ok?

File: backend/api/organisations/resources.py [hunk 2]
```diff
@@ -315,18 +315,6 @@ def get(self):
...
-            try:
-                # Check whether user is admin (can do any query) or user is checking for own projects
```
Comment (willemarcel) [inline backend/api/organisations/resources.py:320]: @ramyaragupathy I removed that restriction in order to be able to query the organizations of another user. Is that ok?

**Comment by ramyaragupathy** [comment]
Just verified this behavior with the current changes - earlier we wanted this to be done only by admin users, now we let anyone even without an access token/login to be able to filter. With login, we provide additional details related to manager list.

An example filter by a manager user ID:

**with authorisation**
```{
  "organisations": [
    {
      "organisationId": 13,
      "name": "OSM Philippines",
      "logo": "https://cdn.hotosm.org/tasking-manager/uploads/1589772584324_512px-OSMPH_Logo.svg.png",
      "description": null,
      "url": null,
      "isManager": null,
      "projects": null,
      "teams": null,
      "campaigns": null
    },
    {
      "organisationId": 1,
      "name": "Other",
      "logo": null,
      "description": null,
      "url": null,
      "isManager": null,
      "projects": null,
      "teams": null,
      "campaigns": null
    }
  ]
}
```

**without authorisation**

```
{
  "organisations": [
    {
      "organisationId": 13,
      "managers": [
        {
          "username": "xxx",
          "pictureUrl": "https://link"
        },
        {
          "username": "xxx",
          "pictureUrl": "https://link"
        }
      ],
      "name": "OSM Philippines",
      "logo": "https://cdn.hotosm.org/tasking-manager/uploads/1589772584324_512px-OSMPH_Logo.svg.png",
      "description": null,
      "url": null,
      "isManager": null,
      "projects": null,
      "teams": null,
      "campaigns": null
    },
    {
      "organisationId": 1,
      "managers": [
        {
          "username": "xxx",
          "pictureUrl": "https://link"
        },
        {
          "username": "xxx",
          "pictureUrl": "https://link"
        }
      ],
      "name": "Other",
      "logo": null,
      "description": null,
      "url": null,
      "isManager": null,
      "projects": null,
      "teams": null,
      "campaigns": null
    }
  ]
}
```
Our current behaviour, we let the manager filter only if the access token is provided. Now when we remove the manager_id check, we let anyone to filter by an user ID without requiring a login. Though we don't provide user's personal details without login, it looks like we're circumventing GDPR compliance by providing a list of user orgs to anyone without an access token. I  believe that we should provide that information only if someone is logged in to access an user profile page. Please let me know your thoughts.

**Comment by willemarcel** [comment]
@ramyaragupathy I agree with you. The intention of my change was to allow any user to see the organizations of other user.

**Comment by willemarcel** [comment]
@ramyaragupathy I made the change to raise 403 error on unauthenticated requests trying to filter orgs by manager_user_id

**Pull request merged**